docs/commit_practives.md

```diff
@@ -0,0 +1,112 @@
+# Commit practices
+
+This document explains the importance and benefits of commit practices and the format of the commit message.
+
+Inspired by and following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
+
+If you have any questions, open the link above.
+
+## Why commit practices are important?
+
+Commit practices are crucial in software development as they ensure a clear and organized history of changes, facilitating collaboration among team members.
+By making frequent, well-documented commits, developers can easily track progress, identify and resolve bugs, and understand the evolution of the codebase.
+Good commit practices enhance code quality and maintainability, enabling efficient version control and seamless integration of contributions.
+
+### Doesn’t this discourage rapid development and fast iteration?
+
+It discourages moving fast in a disorganized way. It helps you be able to move fast long term across multiple projects with varied contributors.
+
+## What are the benefits
+
+- Communicate the nature of changes to teammates, the public, and other stakeholders.
+- Make it easier for people to contribute to your projects, by allowing them to explore a more structured commit history.
+- Automatically generate CHANGELOGs.
+- Automatically determine a semantic version bump (based on the types of commits landed).
+- Trigger build and publish processes.
+
+## Commit format
+
+The commit message should be structured as follows:
+
+```txt
+<type>[optional scope][optional "!"]: <description>
+
+[optional body]
+
+[optional footer(s)]
+```
+
+### Type
+
+The type of commit message is a required parameter and should be one of the following:
+
+- **feat**: A new feature
+- **fix**: A bug fix
+- **docs**: Documentation only changes
+- **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+- **refact**: A code change that neither fixes a bug nor adds a feature
```

Comment:
What about using full word `refactor` here instead of contracted `refact`? Further in the doc in **Tools** section `commitizen` is mentioned; it would be great to use default configuration instead of making each team member keep track of what has changes in the config, follow breaking changes in the tool itself (e.g., config format).

Having a config at all would enable us to add even more entries to it in the future "in case we need it" – it would be great to avoid creating such bureaucracy pitfall. Sure it's easy to add a config entry, but it would increase complexity of onboarding the team to the practice by creating an additional step.

There's [a great article](https://kristoff.it/blog/simple-not-just-easy/) by Loris Cro, Zig core maintainer, that breaks down the "simple-hard vs trivial-complex" dichotomy. Putting it in terms of the article, we want our practice to be trivial and not easy. Speaking in terms of Golang and Python philosophy, no config is better since "less is more".